editor/group_settings_editor.cpp

```diff
@@ -0,0 +1,450 @@
+			edited_scenes_path.push_back(es.path);
+		}
+	}
+	if (!edited_scenes_path.is_empty()) {
+		EditorNode::get_singleton()->save_scene_list(edited_scenes_path);
+		SceneTreeDock::get_singleton()->get_tree_editor()->update_tree();
+	}
+
+	RBSet<String> scenes;
+	_get_all_scenes(EditorFileSystem::get_singleton()->get_filesystem(), scenes);
+	for (RBSet<String>::Element *E = scenes.front(); E; E = E->next()) {
+		Ref<PackedScene> packed_scene = ResourceLoader::load(E->get());
+		if (packed_scene->get_state()->remove_group_references(StringName(p_name))) {
+			ResourceSaver::save(packed_scene, E->get());
+		}
+	}
+}
+
+void GroupSettingsEditor::rename_references(const String &p_old_name, const String &p_new_name) {
+	// Update opened scenes.
+	Vector<String> edited_scenes_path;
+	Vector<EditorData::EditedScene> edited_scenes = EditorNode::get_editor_data().get_edited_scenes();
+	for (const EditorData::EditedScene &es : edited_scenes) {
+		if (!es.root) {
+			continue;
+		}
+		_rename_node_references(es.root, p_old_name, p_new_name);
+		if (!es.path.is_empty()) {
+			edited_scenes_path.push_back(es.path);
+		}
+	}
+	if (!edited_scenes_path.is_empty()) {
+		EditorNode::get_singleton()->save_scene_list(edited_scenes_path);
+		SceneTreeDock::get_singleton()->get_tree_editor()->update_tree();
+	}
+
+	RBSet<String> scenes;
+	_get_all_scenes(EditorFileSystem::get_singleton()->get_filesystem(), scenes);
+	for (RBSet<String>::Element *E = scenes.front(); E; E = E->next()) {
+		Ref<PackedScene> packed_scene = ResourceLoader::load(E->get());
+		if (packed_scene->get_state()->rename_group_references(StringName(p_old_name), StringName(p_new_name))) {
+			ResourceSaver::save(packed_scene, E->get());
+		}
+	}
+}
+
+void GroupSettingsEditor::_remove_node_references(Node *p_node, const String &p_name) {
+	if (p_node->is_in_group(p_name)) {
+		p_node->remove_from_group(p_name);
+	}
+
+	for (int i = 0; i < p_node->get_child_count(); i++) {
+		_remove_node_references(p_node->get_child(i), p_name);
+	}
+}
+
+void GroupSettingsEditor::_rename_node_references(Node *p_node, const String &p_old_name, const String &p_new_name) {
+	if (p_node->is_in_group(p_old_name)) {
+		p_node->remove_from_group(p_old_name);
+		p_node->add_to_group(p_new_name, true);
+	}
+
+	for (int i = 0; i < p_node->get_child_count(); i++) {
+		_rename_node_references(p_node->get_child(i), p_old_name, p_new_name);
+	}
+}
+
+void GroupSettingsEditor::update_groups() {
+	if (updating_group) {
+		return;
+	}
+
+	updating_group = true;
+
+	groups_cache = ProjectSettings::get_singleton()->get_global_groups_list();
+
+	tree->clear();
+
+	TreeItem *root = tree->create_item();
+
+	List<StringName> keys;
+	for (const KeyValue<StringName, String> &E : groups_cache) {
+		keys.push_back(E.key);
+	}
+	keys.sort_custom<NoCaseComparator>();
+
+	for (const StringName &E : keys) {
+		TreeItem *item = tree->create_item(root);
+		item->set_meta("__name", E);
+		item->set_meta("__description", groups_cache[E]);
```

Comment:
What's the purpose of this meta? Isn't the text enough?

Reply:
I use meta in `_item_edited` to compare the old name/description with the new one.